src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,29 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        try {
+            lock.writeLock().lock();
+            list.add(element);
```

Comment:
```suggestion
        lock.writeLock().lock();
        try {
            list.add(element);
```
you can move lock logic outside of try block ;) (the same for other methods)

Reply:
I got it. Thanks!